feat: Add a option to include only plan node headers in plan summary

Rajeev975 opened the pull request:
**Summary:**

- Further helps decrease the summary of a very large query plan.

noHeaderOnly=false

```
-- Project[proj-0]: 11 fields: num_candidates INTEGER, ro_float_features ROW(1450), nro_float_features ROW(1925), ro_id_list_features ROW(167), nro_id_list_features ROW(29), ...
      expressions: call: 44, field: 96
      functions: array_cast_to_float: 43, row_constructor: 1
      projections: 1 out of 11
         raw_label_presences: row_constructor(array_cast_to_float(ROW["raw_label...
      dereferences: 0 out of 11
  -- Project[proj-1]: 11 fields: num_candidates INTEGER, ro_float_features ROW(1450), nro_float_features ROW(1925), ro_id_list_features ROW(167), nro_id_list_features ROW(29), ...
        expressions: call: 1, field: 11
        functions: cast: 1
        projections: 1 out of 11
           raw_label_presences: cast(ROW["model_label_presences"])
        dereferences: 0 out of 11
    -- Project[proj-2]: 11 fields: model_label_presences ROW(43), num_candidates INTEGER, ro_float_features ROW(1450), nro_float_features ROW(1925), ro_id_list_features ROW(167), ...
          expressions: call: 2, constant: 2, field: 11
          functions: cast: 1, padded_make_row_from_map_ROW_6: 1
          constants: ARRAY: 2
          projections: 1 out of 11
             raw_labels: cast(padded_make_row_from_map_ROW_6(ROW["model_flo...
          dereferences: 0 out of 11
      -- Project[proj-3]: 11 fields: model_float_labels MAP, model_label_presences ROW(43), num_candidates INTEGER, ro_float_features ROW(1450), nro_float_features ROW(1925), ...
            expressions: call: 10, constant: 8, field: 26
            functions: cast: 1, coalesce: 8, row_constructor: 1
            constants: ARRAY: 8
            projections: 2 out of 11
               ro_embedding_features: row_constructor(coalesce(ROW["ro_embedding_feature...
               nro_embedding_features: cast(ROW["candidate_embedding_features"])
            dereferences: 0 out of 11
        -- Project[proj-4]: 11 fields: candidate_embedding_features ROW(2), model_float_labels MAP, model_label_presences ROW(43), num_candidates INTEGER, ro_float_features ROW(1450), ...
              expressions: call: 4, constant: 1, field: 12
              functions: cast: 2, coalesce: 1, row_constructor: 1
              constants: ARRAY: 1
              projections: 2 out of 11
                 nro_id_score_list_features: row_constructor(coalesce(cast(ROW["candidate_id_sc...
                 ro_embedding_features: cast(ROW["user_embedding_features"])
              dereferences: 0 out of 11
          -- Project[proj-5]: 11 fields: user_embedding_features ROW(8), candidate_id_score_list_features ROW(1), candidate_embedding_features ROW(2), model_float_labels MAP, model_label_presences ROW(43), ...
                expressions: call: 2, field: 11
                functions: cast: 1, convert_format_ROW_0: 1
                projections: 1 out of 11
                   ro_id_score_list_features: convert_format_ROW_0(cast(ROW["user_id_score_list_...
                dereferences: 0 out of 11
```

noHeaderOnly=true

```
-- Project[proj-0]: 11 fields: num_candidates INTEGER, ro_float_features ROW(1450), nro_float_features ROW(1925), ro_id_list_features ROW(167), nro_id_list_features ROW(29), ...
  -- Project[proj-1]: 11 fields: num_candidates INTEGER, ro_float_features ROW(1450), nro_float_features ROW(1925), ro_id_list_features ROW(167), nro_id_list_features ROW(29), ...
    -- Project[proj-2]: 11 fields: model_label_presences ROW(43), num_candidates INTEGER, ro_float_features ROW(1450), nro_float_features ROW(1925), ro_id_list_features ROW(167), ...
      -- Project[proj-3]: 11 fields: model_float_labels MAP, model_label_presences ROW(43), num_candidates INTEGER, ro_float_features ROW(1450), nro_float_features ROW(1925), ...
        -- Project[proj-4]: 11 fields: candidate_embedding_features ROW(2), model_float_labels MAP, model_label_presences ROW(43), num_candidates INTEGER, ro_float_features ROW(1450), ...
          -- Project[proj-5]: 11 fields: user_embedding_features ROW(8), candidate_id_score_list_features ROW(1), candidate_embedding_features ROW(2), model_float_labels MAP, model_label_presences ROW(43), ...
```

Differential Revision: D77551564

mbasmanova commented on velox/core/PlanNode.h:
```diff
@@ -143,6 +143,14 @@ struct PlanSummaryOptions {
   /// summary.
   size_t maxLength = 50;
 
+  // To provide a simplified, skeleton view of the query plan by removing
```
Try to match the style of the existing comments in this file.

Specifically, 
- use ///
- Remove discussion of plan_summarize_options as it is not available in Velox.

> This will help engineers quickly
>  // understand the plan structure and node hierarchy without being overwhelmed
> // by verbose node details. 

This statement is strange since PlanSummary already aims to reduce the verbosity of the plan. Consider clarifying that this option is needed for extra large plans.

Something like "If true, the output contains just one line per plan node."

Rajeev975 replied:
I have made above changes in latest version of PR

Rajeev975 commented:
> Please, add logic to handle this new option to this PR.

I have made above changes in latest version of PR

Rajeev975 commented:
> @Rajeev975 Would you extend PlanNodeToSummaryStringTest.cpp to test the new boolean?

I have added a new test case in this PR itself.


> @Rajeev975 Would you update PR description to explain how you tested this change? Would be helpful to include examples of plan output with the new option on and off. Thanks.

Added test case and added images before and after implementation of noHeaderOnly variable.

mbasmanova commented on velox/core/PlanNode.cpp:
```diff
@@ -2997,9 +2997,10 @@ void PlanNode::toSummaryString(
-
-  addSummaryDetails(indentation + "      ", options, stream);
-
+  // Check if we need to include details related to plan node.
```
This comment is redundant. Let's remove.

mbasmanova commented on velox/exec/tests/PlanNodeToSummaryStringTest.cpp:
```diff
@@ -150,6 +150,48 @@ TEST_F(PlanNodeToSummaryStringTest, expressions) {
           {.project = {.maxProjections = 3, .maxDereferences = 2}}));
 }
 
+// Below test is to verify that the summary string is truncated when
```
Prefer starting a comment with an active verb: "Verify that ..."

mbasmanova commented on velox/exec/tests/PlanNodeToSummaryStringTest.cpp:
```diff
@@ -150,6 +150,48 @@ TEST_F(PlanNodeToSummaryStringTest, expressions) {
 
+// Below test is to verify that the summary string is truncated when
+// nodeHeaderOnly is set to true.
+TEST_F(PlanNodeToSummaryStringTest, DisplaysNodeHeadersOnly) {
```
DisplaysNodeHeadersOnly -> nodeHeaderOnly

test names should start with lower case

mbasmanova commented on velox/exec/tests/PlanNodeToSummaryStringTest.cpp:
```diff
@@ -96,6 +96,13 @@ TEST_F(PlanNodeToSummaryStringTest, basic) {
           .maxOutputFields = 3,
           .maxChildTypes = 2,
       }));
+  ASSERT_EQ(
```
Please, add empty line before ASSERT_EQ.

Please, add similar checks to other tests.

mbasmanova reviewed (approved):
Thanks.
